server/etcdserver/server.go

```diff
@@ -401,6 +401,10 @@ func NewServer(cfg config.ServerConfig) (srv *EtcdServer, err error) {
 		})
 	}
 
+	// Set the hook after EtcdServer finishes the initialization to avoid
+	// the hook being called during the initialization process.
+	srv.be.SetTxPostLockInsideApplyHook(srv.getTxPostLockInsideApplyHook())
```

Comment:
I'm thinking if there is a better place to install the hook. Putting it on pile with other EtcdServer initialization doesn't seem correct. My thought was that this hook is for CI and should be owned by CI. Have you considered moving the hook installation into `(ci *consistentIndex) SetBackend`?

Reply:
Yes, I thought about moving it into consistentIndex, and it looks more reasonable and clearer. But the reason why I gave up is that we need to make sure the hook is only set after Etcdserver finishes being initialized. The authBatchTx may call LockInsideApply during Etcdserver initialization, the call path is [NewAuthStore](https://github.com/ahrtr/etcd/blob/data_corruption/server/auth/store.go#L944)  --> [authBatchTx.Lock](https://github.com/ahrtr/etcd/blob/4033f5c2b9b57cd341d1b3522abb03bedbeb9a07/server/storage/schema/auth.go#L109) -- > `BatchTx.LockInsideApply`